plugins/synapticsmst/fu-plugin-synapticsmst.c

```diff
@@ -39,18 +37,108 @@ synapticsmst_common_check_supported_system (FuPlugin *plugin, GError **error)
 	return TRUE;
 }
 
+/* creates MST-$str-$BOARDID */
+static void
+fu_plugin_synapticsmst_create_simple_guid (FuDevice *fu_device,
+					   SynapticsMSTDevice *device,
+					   const gchar *str)
+{
+	guint32 board_id = synapticsmst_device_get_board_id (device);
+	g_autofree gchar *guid = g_strdup_printf ("MST-%s-%u", str, board_id);
+
+	fu_device_add_guid (fu_device, guid);
+}
+
+/* creates MST-$str-$chipid-$BOARDID */
+static void
+fu_plugin_synapticsmst_create_complex_guid (FuDevice *fu_device,
+					    SynapticsMSTDevice *device,
+					    const gchar *str)
+{
+	const gchar *chip_id_str = synapticsmst_device_get_chip_id_str (device);
+	g_autofree gchar *chip_id_down = g_ascii_strdown (chip_id_str, -1);
+	g_autofree gchar *tmp = g_strdup_printf ("%s-%s", str, chip_id_down);
+
+	fu_plugin_synapticsmst_create_simple_guid (fu_device, device, tmp);
+}
+
+static gboolean
+fu_plugin_synapticsmst_lookup_device (FuPlugin *plugin,
+				      FuDevice *fu_device,
+				      SynapticsMSTDevice *device,
+				      GError **error)
+{
+	const gchar *board_str;
+	const gchar *guid_template;
+	guint32 board_id = synapticsmst_device_get_board_id (device);
+	const gchar *chip_id_str = synapticsmst_device_get_chip_id_str (device);
+	g_autofree gchar *group = NULL;
+	g_autofree gchar *name = NULL;
+
+	/* GUIDs used only for test mode */
+	if (g_getenv ("FWUPD_SYNAPTICSMST_FW_DIR") != NULL) {
+		g_autofree gchar *tmp = NULL;
+		tmp = g_strdup_printf ("test-%s", chip_id_str);
+		fu_plugin_synapticsmst_create_simple_guid (fu_device, device, tmp);
+		return TRUE;
+	}
+
+	/* Set up the device name (prefer quirks) */
+	group = g_strdup_printf ("SynapticsMSTBoardID=%u", board_id);
+	board_str = fu_plugin_lookup_quirk_by_id (plugin, group,
+					      FU_QUIRKS_NAME);
+	if (board_str == NULL) {
+		if ((board_id >> 8) == CUSTOMERID_DELL)
```

Comment:
I'm a bit confused why we wouldn't have a quirk and we do have code. I think we could just drop the non-quirk case, no?

Reply:
I guess you're right - it would just show up as "unknown platform" then.  The customerid_dell isn't really useful in this case now.